sources/api/pluto/src/main.rs

```diff
@@ -157,9 +192,62 @@ fn get_max_pods(client: &Client, session_token: &str) -> Result<String> {
     error::NoInstanceTypeMaxPods { instance_type }.fail()
 }
 
-fn get_cluster_dns_ip(client: &Client, session_token: &str) -> Result<String> {
+/// Returns the cluster's DNS IPV4 address. First it attempts to call EKS describe-cluster to find
+/// the `serviceIPv4CIDR`. If that works, it returns the first `*.10` address. If the EKS call is
+/// not successful, it falls back to using IMDS MAC CIDR blocks to return one of two default
+/// addresses.
+async fn get_cluster_dns_ip(client: &Client, session_token: &str) -> Result<String> {
+    let region = get_region().await.context(error::Region)?;
+    let cluster_name = get_cluster_name().await.context(error::ClusterName {})?;
```

Comment:
If we fail to get one of these, should we continue on to the default path instead of failing entirely?  It seems like that might be friendlier toward misconfiguration for the majority of cases that won't be using custom CIDR.